staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -32,9 +33,10 @@ type attributesRecord struct {
 	object      runtime.Object
 	oldObject   runtime.Object
 	userInfo    user.Info
+	ae          *audit.Event
 }
 
-func NewAttributesRecord(object runtime.Object, oldObject runtime.Object, kind schema.GroupVersionKind, namespace, name string, resource schema.GroupVersionResource, subresource string, operation Operation, userInfo user.Info) Attributes {
+func NewAttributesRecordWithAudit(object runtime.Object, oldObject runtime.Object, kind schema.GroupVersionKind, namespace, name string, resource schema.GroupVersionResource, subresource string, operation Operation, userInfo user.Info, ae *audit.Event) Attributes {
```

Comment:
I would keep the old func name and make that argument mandatory (possibly nil)

Reply:
OK.
I will modify all `NewAttributesRecord` functions.
```
$ LANG=C grep "NewAttributesRecord" * -r  | grep "\.go"  | wc -l
262
```
:)

Reply:
> 262

https://www.gnu.org/software/sed/manual/html_node/index.html ;)

Reply:
For easier review, please do that in another commit.

99% of those calls are in tests btw. Was surprised that the number is that high.